Tools/CMake/AMReX_SetupCUDA.cmake

```diff
@@ -53,34 +53,34 @@ endif ()
 #
 message(STATUS "Enabled CUDA options:")
 
-set(CUDA_ARCH "Auto" CACHE STRING "CUDA architecture (Use 'Auto' for automatic detection)")
+set(AMReX_CUDA_ARCH "Auto" CACHE STRING "CUDA architecture (Use 'Auto' for automatic detection)")
```

Comment:
In practice, I let my users create a `warpx.profile` in their homes that is sourced to:
- load modules
- set environment hints

for a specific queue of a cluster (e.g. Cori GPU, Cori KNL, Summit, etc.).

For this, it would be super practical if I could hide the architecture in there, too.

I would propose the following logic:
- if an environment variable `AMReX_CUDA_ARCH` is set, take this value as the default
- if no env var is set, perform "Auto"
- if any CMake variable is set, this trumps the "Auto" (as before) and the env var (as it is common on Unix)

Reply:
Sounds good to me. I will update the documentation accordingly.

Reply:
Wuhu, nice! That's very useful for me.